src/main/groovy/io/micronaut/build/pom/MicronautBomExtension.java

```diff
@@ -152,6 +152,20 @@ public interface MicronautBomExtension {
      */
     Property<Boolean> getInferProjectsToInclude();
 
+    /**
+     * Allows setting a list of modules (group:artifact) which are going to be
+     * managed, but for which the version must be inferred from what is in the
+     * catalog. This can be useful when we know a dependency will be used, but
+     * it's not managed by any BOM we include, but it still appears in the
+     * dependency graph.
+     *
+     * The key is `groupId:artifactId` project coordinates and the value is
+     * an alias (without `managed-`) to be included.
```

Comment:
Can we make the alias derived from the artifact id? I find `'org.opentest4j:opentest4j' : 'opentest4j'` a bit verbose

Reply:
We can, however there are risks of collision, and sometimes we want a better alias. For example we have `listenablefuture` for which the alias should be grouped under the `guava` group: `guava-listenablefuture`. Without an explicit alias it would end up at the root of the catalog.

Reply:
Then, can it be made optional? 😇